metaflow/plugins/aws/batch/batch_client.py

```diff
@@ -184,16 +202,21 @@ def __repr__(self):
     def _apply(self, data):
         self._data = data
 
-    @limit(1)
+    @Throttle()
     def _update(self):
         try:
             data = self._client.describe_jobs(jobs=[self._id])
-        except self._client.exceptions.ClientException:
-            return
+        except self._client.exceptions.ClientError as err:
+            code = err.response['ResponseMetadata']['HTTPStatusCode']
+            if code == 429 or code >= 500:
+                raise TriableException(err)
+            raise err
         self._apply(data['jobs'][0])
 
     def update(self):
         self._update()
```

Comment:
The intention here is to be aggressive when we have no data and to silently ignore update errors later on, correct?

Reply:
Exactly. We want to block on the first successful data fetch and afterwards ignore the errors till limits are breached.